src/apps/relay/mainrelay.c

```diff
@@ -3260,24 +3246,32 @@ static void openssl_load_certificates(void)
 	if(!turn_params.no_dtls) {
 #if !DTLS_SUPPORTED
 		TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "ERROR: DTLS is not supported.\n");
-#else
-		if(OPENSSL_VERSION_NUMBER < 0x10000000L) {
+#elif OPENSSL_VERSION_NUMBER < 0x10000000L
 			TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "WARNING: TURN Server was compiled with rather old OpenSSL version, DTLS may not be working correctly.\n");
-		}
-
+#else
+#if OPENSSL_VERSION_NUMBER < 0x10100000L // before openssl-1.1.0 no version independent API
 #if DTLSv1_2_SUPPORTED
-		set_ctx(&turn_params.dtls_ctx,"DTLS",DTLS_server_method());
-		set_ctx(&turn_params.dtls_ctx_v1_2,"DTLS1.2",DTLSv1_2_server_method());
-		SSL_CTX_set_read_ahead(turn_params.dtls_ctx_v1_2, 1);
-		setup_dtls_callbacks(turn_params.dtls_ctx_v1_2);
+		set_ctx(&turn_params.dtls_ctx,"DTLS",DTLSv1_2_server_method()); // openssl-1.0.2
+        if(!turn_params.no_tlsv1_2) {
+            SSL_CTX_set_options(turn_params.dtls_ctx, SSL_OP_NO_DTLSv1_2);
+        }
 #else
-		set_ctx(&turn_params.dtls_ctx,"DTLS",DTLSv1_server_method());
+        set_ctx(&turn_params.dtls_ctx,"DTLS",DTLSv1_server_method()); // < openssl-1.0.2
 #endif
-		SSL_CTX_set_read_ahead(turn_params.dtls_ctx, 1);
+        if(!turn_params.no_tlsv1) {
```

Comment:
what about no-tlsv1_1 ?  it looks like if we are checking only for no_tlsv1_2 and no_tlsv1

Reply:
Right. I think it also how it worked before (there is no DTLS1.1)